### PR TITLE
Add full jacobian for phase change

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- The jacobian matrix for the specific heat was not completely implemented. This reduced the speed of convergence

# Description of the solution

- Implement the full jacobian matrix

# How Has This Been Tested?

- Results of currently tests are not changed. No need to add a new test.

# Comments

- This was much easier than anticipated...

### CODE REVIEW

File: source/solvers/heat_transfer_assemblers.cc [hunk 1]
```diff
@@ -204,6 +204,8 @@ HeatTransferAssemblerBDF<dim>::assemble_matrix(
 
   const std::vector<double> &density       = scratch_data.density;
   const std::vector<double> &specific_heat = scratch_data.specific_heat;
+  const std::vector<double> &grad_specific_heat_temperature =
```
Comment (shahabgol) [inline source/solvers/heat_transfer_assemblers.cc:207]: I don't understand the physical meaning of this parameter.

**Review by shahabgol — COMMENTED** [review]
Clean PR, I just don't understand the meaning of grad_specific_heat_temperature. Maybe you can change the variable name or add a comment to make it clear. :)

File: source/solvers/heat_transfer_assemblers.cc (hunk 1 above)
Comment (blaisb) [reply source/solvers/heat_transfer_assemblers.cc:207]: It is the gradient of the specific heat with respect to the temperature ( so dCp / dT). I'll add a comment to make this clearer.

**Review by shahabgol — APPROVED** [review]
Awesome :)

**Comment by blaisb** [comment]
> Awesome :)

I added a bit more information as a comment to make the thing more clear :). Once CI passes i'll merge.

**Pull request merged**